ikarus/io/resultevaluators.hh

```diff
@@ -86,14 +83,20 @@ struct PrincipalStress
    * @return principal stress
    */
   double operator()(const auto& resultArray, const int comp) const {
-    const auto s_x  = resultArray(0, 0);
-    const auto s_y  = resultArray(1, 0);
-    const auto s_xy = resultArray(2, 0);
+    if constexpr (dim == 2) {
+      const auto s_x  = resultArray(0, 0);
+      const auto s_y  = resultArray(1, 0);
+      const auto s_xy = resultArray(2, 0);
 
-    auto t1 = (s_x + s_y) / 2;
-    auto t2 = std::sqrt(Dune::power((s_x - s_y) / 2, 2) + Dune::power(s_xy, 2));
+      auto t1 = (s_x + s_y) / 2;
+      auto t2 = std::sqrt(Dune::power((s_x - s_y) / 2, 2) + Dune::power(s_xy, 2));
```

Comment:
Can you homogenize this and always use the EigenSolver and `fromVoigt`? The code is the same anyway: https://github.com/FALCONN-LIB/eigen-mirror/blob/ab0a115cf68fd3ba3e471d87f0ec25d6d218d0dc/Eigen/src/Eigenvalues/SelfAdjointEigenSolver.h#L720-L727

Reply:
Ah yes, I tried that, but it actually failed to compile:
```
: error: static assertion failed due to requirement '(int(Eigen::internal::size_of_xpr_at_compile_time<Eigen::Diagonal<Eigen::Matrix<double, 2, 2, 0, 2, 2>, 0>>::ret) == 0 && int(Eigen::internal::size_of_xpr_at_compile_time<Eigen::Block<const Eigen::Matrix<double, 3, 1, 0, 3, 1>, 3, 1, false>>::ret) == 0) || ((int(Diagonal<Eigen::Matrix<double, 2, 2, 0, 2, 2>, 0>::RowsAtCompileTime) == Eigen::Dynamic || int(Block<const Eigen::Matrix<double, 3, 1, 0, 3, 1>, 3, 1, false>::RowsAtCompileTime) == Eigen::Dynamic || int(Diagonal<Eigen::Matrix<double, 2, 2, 0, 2, 2>, 0>::RowsAtCompileTime) == int(Block<const Eigen::Matrix<double, 3, 1, 0, 3, 1>, 3, 1, false>::RowsAtCompileTime)) && (int(Diagonal<Eigen::Matrix<double, 2, 2, 0, 2, 2>, 0>::ColsAtCompileTime) == Eigen::Dynamic || int(Block<const Eigen::Matrix<double, 3, 1, 0, 3, 1>, 3, 1, false>::ColsAtCompileTime) == Eigen::Dynamic || int(Diagonal<Eigen::Matrix<double, 2, 2, 0, 2, 2>, 0>::ColsAtCompileTime) == int(Block<const Eigen::Matrix<double, 3, 1, 0, 3, 1>, 3, 1, false>::ColsAtCompileTime)))': YOU_MIXED_MATRICES_OF_DIFFERENT_SIZES
  EIGEN_STATIC_ASSERT_SAME_MATRIX_SIZE(ActualDstTypeCleaned, Src)
  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/local/include/eigen3/Eigen/src/Core/util/StaticAssert.h:80:3: note: expanded from macro 'EIGEN_STATIC_ASSERT_SAME_MATRIX_SIZE'
  EIGEN_STATIC_ASSERT(EIGEN_PREDICATE_SAME_MATRIX_SIZE(TYPE0, TYPE1), YOU_MIXED_MATRICES_OF_DIFFERENT_SIZES)
  ```

Reply:
I guess the `3`here shouldnt be hard coded: 
```cpp
  Eigen::Matrix<ST, matrixSize, matrixSize> E;
  E.diagonal() = EVoigt.template segment<3>(0);
  ```

Reply:
In `fromVoigt` . Yes this is a bug then. https://github.com/ikarus-project/ikarus/blob/45575521d1e64c6696b5573d9b6b428f734cb6c7/ikarus/utils/tensorutils.hh#L256. Can you replace the three with `matrixSize`.

Reply:
Ah and even better replacing segment with `vector.head<n>();`. The the zero is also not needed.